src/main/java/com/graphhopper/converter/resources/ConverterResource.java

```diff
@@ -1,40 +1,64 @@
 package com.graphhopper.converter.resources;
 
+import com.codahale.metrics.annotation.Timed;
 import com.graphhopper.converter.api.GHResponse;
+import com.graphhopper.converter.api.NominatimResponse;
 import com.graphhopper.converter.core.Converter;
-import com.graphhopper.converter.core.NominatimRequest;
 
 import javax.ws.rs.*;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.core.MediaType;
 import java.util.List;
+import java.util.Optional;
 import java.util.concurrent.atomic.AtomicLong;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
 
 /**
  * @author Robin Boldt
  */
 @Path("/search")
 @Produces(MediaType.APPLICATION_JSON)
-public class ConverterResource
-{
+public class ConverterResource {
 
     private final String nominatimUrl;
     private final AtomicLong counter;
     private final Client jerseyClient;
 
-    public ConverterResource( String nominatimUrl, Client jerseyClient ) {
+    public ConverterResource(String nominatimUrl, Client jerseyClient) {
         this.nominatimUrl = nominatimUrl;
         this.counter = new AtomicLong();
         this.jerseyClient = jerseyClient;
     }
 
+//    @GET
+//    @Produces(MediaType.APPLICATION_JSON)
+//    public List<GHResponse> test(){
+//
+//        NominatimRequest request = new NominatimRequest(jerseyClient);
+//
+//        return Converter.convertFromNominatimList(request.getNominatimResponses());
+//    }
     @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    public List<GHResponse> test(){
+    @Timed
+    public List<GHResponse> handle(@QueryParam("q") String query, @QueryParam("limit") @DefaultValue("5") int limit) {
+        if (limit > 10) {
+            limit = 10;
+        }
 
-        NominatimRequest request = new NominatimRequest(jerseyClient);
+        Response response = jerseyClient.
```

Comment:
The idea of the request was to decouple the parameter handling from the actual NominatimRequest. Don't you think this would be cleaner?